server/modules/analysis/repository/chain/update.js

```diff
@@ -1,26 +1,53 @@
 import { TableChain } from '../../../../../common/model/db'
 
 import { db } from '../../../../db/db'
+import * as DbUtils from '../../../../db/dbUtils'
 
 /**
- * Updates the common script of the given processing chain.
+ * Updates the fields passed as argument of a processing chain.
  *
  * @param {!object} params - The query parameters.
  * @param {!string} params.surveyId - The survey id.
  * @param {!string} params.chainUuid - The processing chain uuid.
- * @param {!string} params.scriptCommon - The script content.
+ * @param {!object<string, any>} params.fields - A <key, value> object containing the fields to update.
  * @param {pgPromise.IDatabase} [client=db] - The database client.
  *
  * @returns {Promise<null>} - The result promise.
  */
-export const updateChainScriptCommon = (params, client = db) => {
-  const { surveyId, chainUuid, scriptCommon } = params
+export const updateChain = (params, client = db) => {
+  const { surveyId, chainUuid, fields } = params
   const table = new TableChain(surveyId)
 
   return client.none(
     `UPDATE ${table.nameQualified}
-    SET ${TableChain.columnSet.scriptCommon} = $2
+    SET ${Object.keys(fields)
+      .map((field, i) => `${field} = $${i + 2}`)
+      .join(', ')}
     WHERE ${TableChain.columnSet.uuid} = $1`,
-    [chainUuid, scriptCommon]
+    [chainUuid, ...Object.values(fields)]
+  )
+}
+
+/**
+ * Updates the processing chain execution status.
+ *
+ * @param {!object} params - The query parameters.
+ * @param {!string} params.surveyId - The survey id.
+ * @param {!string} params.chainUuid - The processing chain uuid.
+ * @param {!string} params.statusExec - The execution status.
+ * @param {pgPromise.IDatabase} [client=db] - The database client.
+ *
+ * @returns {Promise<null>} - The result promise.
+ */
+export const updateChainStatusExec = (params, client = db) => {
+  const { surveyId, chainUuid, statusExec } = params
+  const table = new TableChain(surveyId)
+
+  return client.none(
```

Comment:
why don't you call `updateChain` instead?

Reply:
is this outdated? I changed it while doing self review